### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by ArtemKlots**

### CODE REVIEW

File: src/main/java/com/travelBill/telegram/TelegramBot.java [hunk 1]
```diff
@@ -15,18 +15,19 @@
...
     }
 
     @Override
     public void onUpdateReceived(Update update) {
         User currentUser = setupUser(update);
-        SendMessage message = actionRouter.delegateAction(update, currentUser);
+        SendMessage message = scenarioFactory.getScenario(update, currentUser).perform();
 
         try {
             execute(message);
```
Comment (EduardRakov) [inline src/main/java/com/travelBill/telegram/TelegramBot.java:34]: Look at https://github.com/rainerhahnekamp/sneakythrow

Comment (EduardRakov) [reply src/main/java/com/travelBill/telegram/TelegramBot.java:34]: - Use methods from Update e.g. hasMessage() etc.
- Throw and handle the case when no message or...

Comment (EduardRakov) [reply src/main/java/com/travelBill/telegram/TelegramBot.java:34]:
```
private User setupUser(Update update) {
        org.telegram.telegrambots.meta.api.objects.User u;
        if (update.hasMessage()) {
            u = update.getMessage().getFrom();

        } else if (update.hasCallbackQuery()) {
            u = update.getCallbackQuery().getFrom();
        }

        return telegramUserService.setupUser(u);
    }
```

File: src/main/java/com/travelBill/telegram/scenario/ScenarioFactory.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    private final EventService eventService;
+
+    public ScenarioFactory(EventService eventService) {
+        this.eventService = eventService;
+    }
+
+    public Scenario getScenario(Update update, User currentUser) {
+        ScenarioTypes type = ScenarioTypes.defineType(update);
+
+        if (type == null) {
```
Comment (EduardRakov) [inline src/main/java/com/travelBill/telegram/scenario/ScenarioFactory.java]: use isNull() instead

File: src/main/java/com/travelBill/telegram/scenario/event/EventScenarioFactory.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package com.travelBill.telegram.scenario.event;
+
+import com.travelBill.telegram.scenario.UnknownScenario;
+import com.travelBill.telegram.scenario.common.Scenario;
+
+public class EventScenarioFactory {
+    public Scenario getScenario(EventContext eventContext) {
+        EventActions type = EventActions.defineType(eventContext.update);
+        if (type == null) {
```
Comment (EduardRakov) [inline src/main/java/com/travelBill/telegram/scenario/event/EventScenarioFactory.java]: DRY

**Review by EduardRakov — CHANGES_REQUESTED** [review]

**Comment by ArtemKlots** [comment]
@EduardRakov ready for review

**Review by EduardRakov — APPROVED** [review]

**Pull request merged**